lib/matplotlib/dates.py

```diff
@@ -941,7 +947,10 @@ def datalim_to_dt(self):
         dmin, dmax = self.axis.get_data_interval()
         if dmin > dmax:
             dmin, dmax = dmax, dmin
-
+        if dmin < 1:
+            raise ValueError('datalim minimum {} < 1. This often '
+                             'happens if you pass a float to an '
```

Comment:
Can "float" be change to something less technical like "number" or "normal number"?

Reply:
None datetime value?